### PR TITLE
[ELY-2063] Masked password interface contains 8 algorithms unavailable on IBM JDK 8

### PR DESCRIPTION
**Pull request by SoniaZaldana**
https://issues.redhat.com/browse/ELY-2063

### CODE REVIEW

File: credential/base/src/main/java/org/wildfly/security/password/interfaces/MaskedPassword.java [hunk 1]
```diff
@@ -41,13 +41,6 @@
...
     String ALGORITHM_MASKED_SHA1_DES_EDE = "masked-SHA1-DES-EDE";
     String ALGORITHM_MASKED_SHA1_DES_EDE_CBC_PKCS5 = "masked-SHA1-DES-EDE-CBC-PKCS5";
-    String ALGORITHM_MASKED_SHA1_RC2_40 = "masked-SHA1-RC2-40";
-    String ALGORITHM_MASKED_SHA1_RC2_40_CBC_PKCS5 = "masked-SHA1-RC2-40-CBC-PKCS5";
-    String ALGORITHM_MASKED_SHA1_RC2_128 = "masked-SHA1-RC2-128";
-    String ALGORITHM_MASKED_SHA1_RC2_128_CBC_PKCS5 = "masked-SHA1-RC2-128-CBC-PKCS5";
-    String ALGORITHM_MASKED_SHA1_RC4_40 = "masked-SHA1-RC4-40";
-    String ALGORITHM_MASKED_SHA1_RC4_40_ECB = "masked-SHA1-RC4-40-ECB";
-    String ALGORITHM_MASKED_SHA1_RC4_128 = "masked-SHA1-RC4-128";
     String ALGORITHM_MASKED_SHA1_RC4_128_ECB = "masked-SHA1-RC4-128-ECB";
```
Comment (fjuma) [inline credential/base/src/main/java/org/wildfly/security/password/interfaces/MaskedPassword.java:44]: Is this one unsupported as well?

Comment (SoniaZaldana) [reply credential/base/src/main/java/org/wildfly/security/password/interfaces/MaskedPassword.java]: Yes, sorry I missed it. I moved it.

**Review by fjuma — APPROVED** [review]

File: password/impl/src/test/java/org/wildfly/security/password/impl/MaskedPasswordTest.java [hunk 2]
```diff
@@ -79,6 +80,9 @@ public static void removeProvider() {
 
     @Test
     public void testClearSpec() throws Exception {
+        Assume.assumeFalse("IBM JDK does not support some of the algorithms used in this test. See ELY-2063.",
```
Comment (marekkopecky) [inline password/impl/src/test/java/org/wildfly/security/password/impl/MaskedPasswordTest.java:83]: Is necessary to ignore all algorithms on IBM JDK in this test? Many of them are working.

Comment (fjuma) [reply password/impl/src/test/java/org/wildfly/security/password/impl/MaskedPasswordTest.java:83]: Good point, we should make sure that we only ignore the unsupported algorithms when running on IBM JDK.

**Review by fjuma — APPROVED** [review]

**Review by darranl — APPROVED** [review]

**Pull request merged**